File: backend/app/schemas/compound.py
from dataclasses import dataclass, field
from typing import Literal, Optional


@dataclass
class AnalyzeOptions:
    radius: int = 2
    n_bits: int = 2048
    use_features: bool = False
    top_n: int = 5


@dataclass
class CompoundAnalyzeRequest:
    input_type: Literal["smiles", "name", "cas", "mol"]
    input_value: str
    options: Optional[AnalyzeOptions] = None


@dataclass
class SimilarityResult:
    reference_name: str
    reference_cas: str
    similarity: float
    rank: int


@dataclass
class CompoundAnalyzeResponse:
    compound_name: Optional[str]
    canonical_smiles: str
    total_score: int
    base_score: float
    composite_score: float
    top_similarities: list[SimilarityResult] = field(default_factory=list)
    algorithm_version: str = ""
    fingerprint_params: dict[str, object] = field(default_factory=dict)
    cached: bool = False
    calc_id: str = ""

File: frontend/src/types/compound.ts
export interface AnalyzeOptions {
  radius?: number;
  n_bits?: number;
  use_features?: boolean;
  top_n?: number;
}

export interface SimilarityResult {
  reference_name: string;
  reference_cas: string;
  similarity: number;
  rank: number;
}

export interface CompoundAnalyzeRequest {
  input_type: "smiles" | "name" | "cas" | "mol";
  input_value: string;
  options?: AnalyzeOptions;
}

export interface CompoundAnalyzeResponse {
  compound_name: string | null;
  canonical_smiles: string;
  total_score: number;
  base_score: number;
  composite_score: number;
  top_similarities: SimilarityResult[];
  algorithm_version: string;
  fingerprint_params: Record<string, unknown>;
  cached: boolean;
  calc_id: string;
}
